### PR TITLE
httplib: Adjust for our needs in http_c and ncch crypto support

### PR DESCRIPTION
**Pull request by B3n30**
The implementation of http_c and other parts of citra (like crypto) needs more features then httplib currently supports. This PR add those features we currently know of are necessary. 

I will also try to upstream these changes to the httplib repo, but for now I think it's best to have those changes in our repo so we can continue with our development (e.g. as far as i know the ncch crypto Pr from wwylele just waits for that).

- Add data to progress callback to fetch data while still downloading 
- add set_verify() to set the verification flags for ssl
- add_client_cert_ASN1() to add client certs to an ssl context
- add_cert() to add root certs to an ssl context

---
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/4044)

### CODE REVIEW

File: externals/httplib/httplib.h [hunk 1]
```diff
@@ -100,14 +100,21 @@ struct ci {
...
+
 typedef std::multimap<std::string, std::string, detail::ci>  Headers;
 
 template<typename uint64_t, typename... Args>
 std::pair<std::string, std::string> make_range_header(uint64_t value, Args... args);
 
 typedef std::multimap<std::string, std::string>                Params;
 typedef std::smatch                                            Match;
-typedef std::function<void (uint64_t current, uint64_t total)> Progress;
+typedef std::function<void (uint64_t current, uint64_t total, std::string data)> Progress;
```
Comment (wwylele) [inline externals/httplib/httplib.h]: `const std::string&` so that it won't copy the entire data buffer every time it updates progress.

Comment (B3n30) [reply externals/httplib/httplib.h]: I made a mistake, that header and body is already accessible through the result. So I removed the data from Progress again. I still call the progress callback with zero length on read_content_without_length

**Comment by B3n30** [comment]
The callback call for read_content_chunked is still missing, but that shouldn't be an issue for citra

**Comment by wwylele** [comment]
@B3n30 are you going to change the README?

**Comment by B3n30** [comment]
I would like to put this to an halt since my recent research hinted that parts of this PR aren't necessary. In particular: `Add data to progress callback to fetch data while still downloading`. But I'm still not 100% sure if or if not.

**Comment by bunnei** [comment]
@B3n30 can you please provide an update here?

**Comment by B3n30** [comment]
Closing this since I will just introduce the changes when they are needed in the HTTP_C Service implementation

**Pull request closed without merging**